simpleclient_logback/src/main/java/io/prometheus/client/logback/InstrumentedAppender.java

```diff
@@ -0,0 +1,84 @@
+package io.prometheus.client.logback;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.UnsynchronizedAppenderBase;
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.Counter;
+
+public class InstrumentedAppender extends UnsynchronizedAppenderBase<ILoggingEvent> {
+
+  public static final String COUNTER_NAME = "logback_appender_total";
+
+  private final CollectorRegistry registry;
+
+  private Counter counter;
+  private Counter.Child trace;
+  private Counter.Child debug;
+  private Counter.Child info;
+  private Counter.Child warn;
+  private Counter.Child error;
+
+  /**
+   * Create a new instrumented appender using the default registry.
+   */
+  public InstrumentedAppender() {
+    this(CollectorRegistry.defaultRegistry);
+  }
+
+  /**
+   * Create a new instrumented appender using the given registry.
+   *
+   * @param registry the collector registry
+   */
+  public InstrumentedAppender(CollectorRegistry registry) {
+    this.registry = registry;
+    setName(Appender.class.toString());
+  }
+
+  @Override
+  public void start() {
```

Comment:
We also have multiple of a given type of Appender configured in some of our services with different parameters, which I can only see working with multiple objects. I also don't see any restriction otherwise mentioned as part of the interface.
I see no code in the AppenderBase to prevent multiple calls to start(), and there's no statics to protect against multiple instances existing. The protections are on doAppend(), rather than on the object lifecycle.

As we need to be absolutely certain that this code is run only once, I think it's safest to do the statics ourself.